server/testutil/testbes/testbes.go

```diff
@@ -109,3 +214,26 @@ func Run(t testing.TB) (*TestBuildEventServer, pepb.PublishBuildEventClient) {
 
 	return server, client
 }
+
+// RunTCP starts a test BES server on a random TCP port and returns the server and its address.
+// This is useful for testing clients that need to dial a real address.
+// The returned address is prefixed with "grpc://" to indicate an insecure connection.
+func RunTCP(t testing.TB) (*TestBuildEventServer, string) {
```

Comment:
is the `Run` function needed anymore? wonder if we can just update that implementation to use TCP instead, since it's still using loopback networking anyway and should be fast/low overhead, should guaranteed to have no port collisions (due to `:0`), and is more realistic since we use TCP in the real app.